app/javascript/chat/ReportAbuse/index.jsx

```diff
@@ -0,0 +1,173 @@
+import { h } from 'preact';
+import PropTypes from 'prop-types';
+import { useState } from 'preact/hooks';
+import { reportAbuse, blockUser } from '../actions/requestActions';
+import { addSnackbarItem } from '../../Snackbar';
+import { Button, FormField, RadioButton } from '@crayons';
+
+/**
+ * This component render the report abuse
+ *
+ * @param {object} props
+ * @param {object} props.data
+ * @param {function} props.closeReportAbuseForm
+ *
+ * @component
+ *
+ * @example
+ *
+ * <ReportAbuse
+ *  data={data}
+ *  closeReportAbuseForm={closeReportAbuseForm}
+ * />
+ *
+ */
+function ReportAbuse({ data, closeReportAbuseForm }) {
+  const [category, setCategory] = useState(null);
+
+  const handleChange = (e) => {
+    setCategory(e.target.value);
+  };
+
+  const handleSubmit = async () => {
+    const response = await reportAbuse(
+      data.message,
+      'connect',
+      category,
+      data.user_id,
+    );
+    const { success, message } = response;
+    if (success) {
+      const confirmBlock = window.confirm(
+        `Are you sure you want to block this person? This will:
+      - prevent them from commenting on your posts
+      - block all notifications from them
+      - prevent them from messaging you via DEV Connect`,
+      );
+      if (confirmBlock) {
+        const response = await blockUser(data.user_id);
+        if (response.result === 'blocked') {
+          addSnackbarItem({
+            message: 'Your report has been submitted and User has been blocker',
+          });
+        }
+      } else {
+        addSnackbarItem({ message: 'Your report has been submitted.' });
+      }
+      closeReportAbuseForm();
+    } else {
+      addSnackbarItem({ message });
+    }
+  };
+
+  return (
+    <div>
+      <section className="p-4 grid gap-2 crayons-card mb-4">
+        <h1 className="lh-tight mb-4 mt-0">Report Abuse</h1>
+        <p>
+          Thank you for reporting any abuse that violates our{' '}
+          <a href="/code-of-conduct">code of conduct</a> or
+          <a href="/terms">terms and conditions</a>. We continue to try to make
+          environment a great one for everybody.
+        </p>
+      </section>
+      <section className="crayons-card crayons-card--secondary p-4 justify-between">
+        <FormField variant="radio">
```

Comment:
Do these radio buttons get rendered in a `fieldset`? That would help to group them.

Reply:
Good catch @marcysutton, they do not render in a [fieldset](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/fieldset). @narender2031, can you make the change Marcy is suggesting?

Reply:
done